### PR TITLE
Add Intune Firewall rules policy for Windows10 ConfigMgr

### PR DESCRIPTION
**Pull request by FabienTschanz**
#### Pull Request (PR) description
This PR adds the `IntuneFirewallRulesPolicyWindows10ConfigMgr` resource. This is the equivalent of the `IntuneFirewallRulesPolicyWindows10`, but for the configuration manager aka SCCM. 

@NikCharlebois Can you give some feedback to the responsible product group that there is no way to modify an existing policy because the UI is not correctly working? It's showing the design of an /intents policy instead of a /configurationPolicies policy. The `Configuration settings` section stays empty, even though there are settings configured. 

image

#### This Pull Request (PR) fixes the following issues
None.

### CODE REVIEW

File: Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/3-Remove.ps1 [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<#
+This example removes a Device Control Policy.
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/3-Remove.ps1]: This description looks incorrect

File: Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/2-Update.ps1 [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<#
+This example updates a Intune Firewall Policy for Windows10.
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/2-Update.ps1]: Does it make sense to add "via Configuration Manager" at the end of the line?

File: Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/1-Create.ps1 [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<#
+This example creates a new Intune Firewall Policy for Windows10.
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/1-Create.ps1]: Does it make sense to add "via Configuration Manager" at the end of the line?

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1 [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    try
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1:69]: Please add a Write-Verbose line that says this is the beginning of the Get method, like is done here:
https://github.com/microsoft/Microsoft365DSC/blob/ba9e95d9942dc58cea783af8d4db89ff31ec6a5c/Modules/Microsoft365DSC/DSCResources/MSFT_AADAuthenticationMethodPolicyAuthenticator/MSFT_AADAuthenticationMethodPolicyAuthenticator.psm1#L68

That way it is easy to track what function the code is in.

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1 [hunk 5]
```diff
@@ -0,0 +1,641 @@
...
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity,
+
+        [Parameter()]
+        [System.String[]]
+        $AccessTokens
+    )
+
+    #Ensure the proper dependencies are installed in the current environment.
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1:247]: Same comment as for the Get method:
https://github.com/microsoft/Microsoft365DSC/blob/ba9e95d9942dc58cea783af8d4db89ff31ec6a5c/Modules/Microsoft365DSC/DSCResources/MSFT_AADAuthenticationMethodPolicyAuthenticator/MSFT_AADAuthenticationMethodPolicyAuthenticator.psm1#L566

**Review by ykuijs — CHANGES_REQUESTED** [review]
Just a few comments.

Would it be possible to fix these before 14:00h CET (GMT+1) tomorrow? After that I will start working on the new release and it would be great if we could include this resource. No worries, if that is not possible.....not trying to rush you 😉

**Comment by ykuijs** [comment]
Also, please merge with Dev

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1 (hunk 5 above)
Comment (FabienTschanz) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1:247]: I'm using the default template of the resource generator and there is no line like that in pretty much all of the Intune resources. Do you still want me to add it?
Edit: What do we do with the other Intune resources? Updating them as well?

File: Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/3-Remove.ps1 (hunk 1 above)
Comment (FabienTschanz) [reply Modules/Microsoft365DSC/Examples/Resources/IntuneFirewallRulesPolicyWindows10ConfigMgr/3-Remove.ps1]: It sure is... That's what I get from copy pasting 😆

File: Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1 (hunk 5 above)
Comment (ykuijs) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1:247]: That is good to know. We need to update the DRG as well. Please update this resource.

It is on my ToDo list to go through all resources to make sure they all have these basics. So if you run into instances, please correct.....never hurts to have a little help checking and updating so many resources 😉

Comment (FabienTschanz) [reply Modules/Microsoft365DSC/DSCResources/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr/MSFT_IntuneFirewallRulesPolicyWindows10ConfigMgr.psm1:247]: Will do 👍

**Comment by FabienTschanz** [comment]
@ykuijs The resource is now updated with the Get- and Set- message as requested. An entry in the resource generator was also added, so each new resource created from that will include the `Write-Verbose` messages with their primary and filter key.

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**